src/controllers/events.js

```diff
@@ -29,3 +29,18 @@ exports.getRelatedEvents = async (req, res, next) => {
     .populate('ngo', 'name');
   return res.status(200).json({ success: true, data: relatedEvents });
 };
+
+exports.applyToEvent = async (req, res, next) => {
+  const volunteer = await Volunteer.findById(req.user._id);
+  const event = await Event.findById(req.params.id);
+
+  if (!event) {
+    return next(new ErrorResponse('No event found', 404));
+  }
+  event.pendingApplicants.push(req.user._id);
+  await event.save();
+
+  volunteer.appliedEvents.push(req.params.id);
```

Comment:
The event would be added to the `volunteer.appliedEvents` only when he's approved.